packages/eas-cli/src/metadata/context.ts

```diff
@@ -0,0 +1,110 @@
+  platform: Platform.IOS;
+  /** Submission profile name to use */
+  profile: IosSubmitProfile;
+  /** Configured store configuration file name (defaults to store.config.json) */
+  metadataFile: string;
+  /** Authenticated Expo account */
+  user: Actor;
+  /** The store credentials manager */
+  credentialsCtx: CredentialsContext;
+  /** The app bundle identifier to use for the stores */
+  bundleIdentifier: string;
+  /** Root of the Expo project directory */
+  projectDir: string;
+  /** Resolved Expo app configuration */
+  exp: ExpoConfig;
+  /**
+   * If we should run in non-interactive mode, as much as we can.
+   * The store authentication requires a fully configured ASC environment, or interactive mode.
+   */
+  nonInteractive: boolean;
+};
+
+export type MetadataAppStoreAuthentication = {
+  /** The root entity of the App store  */
+  app: App;
+  /** The authentication state we used to fetch the root entity */
+  auth: Partial<Session.AuthState>;
+};
+
+/**
+ * Metadata uses the submission profile to find the configured metadata filename.
+ * Note, only iOS is supported for metadata right now.
+ */
+export async function createMetadataContextAsync(params: {
+  projectDir: string;
+  exp: ExpoConfig;
+  profileName?: string;
+  nonInteractive?: boolean;
+}): Promise<MetadataContext> {
+  const submissionProfiles = await getProfilesAsync({
+    type: 'submit',
+    platforms: [Platform.IOS],
+    projectDir: params.projectDir,
+    profileName: params.profileName,
+  });
+
+  const submissionProfile = submissionProfiles.find(profile => profile.platform === Platform.IOS);
+  assert(
+    submissionProfile,
+    'Could not resolve the iOS submission profile, only iOS is supported for metadata'
+  );
+  const iosSubmissionProfile = submissionProfile.profile as IosSubmitProfile;
+
+  const exp = params.exp ?? getExpoConfig(params.projectDir);
+  const user = await ensureLoggedInAsync({ nonInteractive: params.nonInteractive });
+  const bundleIdentifier = await getBundleIdentifierAsync(params.projectDir, exp);
+
+  const credentialsCtx = new CredentialsContext({
+    user,
+    exp: params.exp,
+    nonInteractive: params.nonInteractive,
+    projectDir: params.projectDir,
+  });
+
+  return {
+    platform: Platform.IOS,
+    profile: iosSubmissionProfile,
+    metadataFile: iosSubmissionProfile.meta ?? 'store.config.json',
+    user,
+    credentialsCtx,
+    bundleIdentifier,
+    projectDir: params.projectDir,
+    exp: params.exp,
+    nonInteractive: params.nonInteractive ?? false,
+  };
+}
+
+/**
+ * To start syncing ASC entities, we need access to the apple utils App instance.
+ * This ensures we are authenticated and have the app instance ready to use.
+ */
+export async function ensureMetadataAppStoreAuthenticatedAsync({
+  credentialsCtx,
+  bundleIdentifier,
+}: MetadataContext): Promise<MetadataAppStoreAuthentication> {
+  const authCtx = await credentialsCtx.appStore.ensureAuthenticatedAsync();
```

Comment:
how are you handling nonInteractive option here? ensureAuthenticatedAsync is interactive

Reply:
It can't. None of the `AppStoreApi` methods from the `CredentialsContext` can be set to non-interactive, that's why I added a warning here: https://github.com/expo/eas-cli/blob/%40bycedric/metadata/context/packages/eas-cli/src/metadata/context.ts#L32-L35

Preferably, we would just pass through the `nonInteractive` mode set on the `CredentialsContext` here: https://github.com/expo/eas-cli/blob/%40bycedric/metadata/context/packages/eas-cli/src/credentials/context.ts#L19

But, that would change existing functionality. We can still do that, but I guess in a different PR. For now, it would fail in the `prompts` method when authenticating in CI without the proper envvars. If people configured the right envvars, it should not go into an interactive mode (should pick up the right envvars and just authenticate with that directly).

Reply:
If there is no way to do that in nonInteractive mode then there should be no nonInteractive option

Reply:
@wkozyra95 would you be ok with commenting the `nonInteractive` option out, but still add a `todo` to implement that later in the `AppStoreApi`?